core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -40,13 +42,25 @@ class AdminClient(val time: Time,
                   val client: ConsumerNetworkClient,
                   val bootstrapBrokers: List[Node]) extends Logging {
 
+  @volatile var running: Boolean = true
+
+  val networkThread = new KafkaThread("admin-client-network-thread", new Runnable {
+    override def run() {
+      while (running) {
+        client.poll(Long.MaxValue)
+        // Allow user thread to get the lock on the ConsumerNetworkClient to enqueue requests.
+        this.synchronized()
```

Comment:
It is actually needed as a way for user thread to get the lock to enqueue requests once sender thread has finished its current `client.poll(...)`. Even though it is empty, this synchronzed block is still useful because some other thread may have the lock here.

This trick will no longer be needed after KAFKA-4820 is committed. Do you have time to take a look at https://github.com/apache/kafka/pull/2619?